controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +201,76 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context, state *consumerReconcileState) error {
+	r.keyIndex.mu.Lock()
+	defer r.keyIndex.mu.Unlock()
+
+	// Clear old indexes
+	r.keyIndex.index = make(map[string]map[string]map[string]string)
+
+	// Check conflicts for all valid consumers in the current state
+	for ns, consumers := range state.namespaceToConsumers {
+		for _, consumer := range consumers {
+			if err := r.indexConsumer(ns, consumer); err != nil {
+				return err
```

Comment:
Instead of returning err from the Reconcile, we should set the status of Consumer and remove it from the state, as https://github.com/mosn/htnn/blob/b378583022889607d57534b9d4c674e3e9fdb4df/controller/internal/controller/consumer_controller.go#L115.

Return err in Reconcile will trigger a retry, which doesn't help the situation.